staker/block_validator.go

```diff
@@ -507,26 +507,20 @@ func (v *BlockValidator) createNextValidationEntry(ctx context.Context) (bool, e
 			return false, err
 		}
 		// Check that Espresso block numbers increase consecutively. This ensures that the sequenced L2 chain does not skip an Espresso block.
-		var prevEspressoMsg *arbostypes.L1IncomingMessage
+		var prevEspressoJst *arbostypes.EspressoBlockJustification
 		for i := pos - 1; i != 0; i-- {
 			msg, err := v.streamer.GetMessage(i)
 			if err != nil {
 				return false, err
 			}
-			if msg.Message.Header.Kind == arbostypes.L1MessageType_L2Message {
-				prevEspressoMsg = msg.Message
-				break
-			}
-		}
-		if prevEspressoMsg != nil {
-			_, prevJst, err := arbos.ParseEspressoMsg(prevEspressoMsg)
+			_, prevJst, err := arbos.ParseEspressoMsg(msg.Message)
 			if err != nil {
-				return false, err
-			}
-			if prevJst.Header.Height+1 != jst.Header.Height {
-				return false, fmt.Errorf("l2 chain appears to have skipped an espresso block, last espresso block number: %d, current: %d", prevJst.Header.Height, jst.Header.Height)
+				prevEspressoJst = prevJst
 			}
 		}
+		if prevEspressoJst != nil && prevEspressoJst.Header.Height+1 != jst.Header.Height {
```

Comment:
If `prevEspressoJst == nil` do we need to check that the height is equal to the genesis height or something? Or is that guaranteed because if there was a previous justification we would have been guaranteed to find it?